### DISCUSSION TITLE
Some questions about special phrases

### DISCUSSION THREAD

**Discussion by Juanjo-MC** [osm-search/Nominatim#2477] [Q&A]
Hi,

I've recently modified [Spanish special phrases](https://wiki.openstreetmap.org/wiki/Nominatim/Special_Phrases/ES) and it seems that they have been loaded into nominatim now.
Among several corrections, I added several new sections, some are working as expected, others not so much, may be you can explain what I did wrong.

I added a 'Comer' (eat/eating) section and created the phrases for amenity=restaurant, amenity=fast_food and food=yes. This last one is to also find pubs that offers food services, in Spain most of them do.
It is working for restaurants and fast foods but it never find pubs with food=yes. I guess that this tag is not indexed.

I also added a 'Cajeros automáticos' (ATM) section looking for atm=yes to find ATMs that are mapped as amenity=bank + atm=yes. It doesn't work, I guess for the same reason as above.

I modified an existing one for alpine huts, the old one was saying 'Cabañas alpinas' which is the literal translation from English but not the way they are named in Spain, I changed it to 'Refugios de montaña'. That one isn't working at all, I don't know why.

Finally, I created an 'Alojamiento/Alojarse' (hosting) section looking for hotels, camp sites, guest houses, ... Strangely, it doesn't work when using special phrases ([example](https://nominatim.openstreetmap.org/search?q=alojamiento%20en%20cabrales&format=jsonv2&addressdetails=1&extratags=1&namedetails=1&limit=50)) but it works when a bounding box is used ([example](https://nominatim.openstreetmap.org/search?q=alojamiento&format=jsonv2&addressdetails=1&extratags=1&namedetails=1&limit=50&bounded=1&viewbox=-4.824157,43.311689,-4.791756,43.294294)).

Thanks.

**Comment by lonvia** [comment]
There is no support for secondary tags like atm=yes and food=yes. This would require a completely different implementation on the Nominatim side. I have removed those entries from the page before reimporting the phrases.

I have also removed the generic terms 'comer' and 'alojarse' that you have added. Again, this kind of generic category search is not something that was intended and the implementation is not really ready for it.

The handling of special phrases of near/in type was broken. This is fixed now and the special phrases are reimported.

I still need to figure out what is going on with the alpine hut.

**Reply by Juanjo-MC** [reply]
> There is no support for secondary tags like atm=yes and food=yes. This would require a completely different implementation on the Nominatim side. I have removed those entries from the page before reimporting the phrases.

understood.

> I have also removed the generic terms 'comer' and 'alojarse' that you have added. Again, this kind of generic category search is not something that was intended and the implementation is not really ready for it.

OK. Is this something that you want/plan to implement? IMO adding such kind of generic search capability to Nominatim would be a great improvement.

Thanks

**Reply by lonvia** [reply]
I agree that both features (secondary tags and generic categories) would be really nice to have. So, yes, we might have that one day. It is simply a question of priorities and for now the main focus is still on finding place names.

**Reply by Juanjo-MC** [reply]
Ok, thanks 👍

**Comment by lonvia** [comment]
I've fixed the issue with the 'Refugios de montaña' now.

**Reply by Juanjo-MC** [reply]
well, It's _**partially**_ fixed, for example, searching for alpine huts in Cabrales only the ['refugios de montaña cerca de cabrales'](https://nominatim.openstreetmap.org/ui/search.html?q=refugios+de+monta%C3%B1a+cerca+de+cabrales&limit=50) (alpine huts near ...) variant is returning results.
Serching alpine huts in Onís, where there is [one](https://www.openstreetmap.org/way/126231110), is not returning any result.

**Reply by lonvia** [reply]
That's because Nominatim doesn't consider the hut to be near Onis (or in Onis). It only considers POIs near the center point of the place in question. At this point, I repeat the usual disclaimer: Nominatim is not a tool for downloading complete lists of POIs for large areas. I recommend using Overpass for that.

**Reply by Juanjo-MC** [reply]
> That's because Nominatim doesn't consider the hut to be near Onis (or in Onis). It only considers POIs near the center point of the place in question

Understood, thank you lonvia.
